File: leetcode/#836.py
# question --> https://leetcode.com/problems/rectangle-overlap/

class Solution:
    def isRectangleOverlap(self, rec1: List[int], rec2: List[int]) -> bool:
        if rec1[3] <= rec2[1] or rec1[1] >= rec2[3] or rec1[2] <= rec2[0] or rec1[0] >= rec2[2]:
            return 0
        return 1
        
        # TC: O(1); SC: O(1)
        
# end 

File: leetcode/#836.js
// question --> https://leetcode.com/problems/rectangle-overlap/

/**
 * @param {number[]} rec1
 * @param {number[]} rec2
 * @return {boolean}
 */
var isRectangleOverlap = function(rec1, rec2) {
    if (rec1[3] <= rec2[1] || rec1[1] >= rec2[3] || rec1[2] <= rec2[0] || rec1[0] >= rec2[2]) {
        return 0
    }
    return 1
};
        
// TC: O(1); SC: O(1)

// end 